src/Microsoft.Diagnostics.Monitoring.WebApi/EgressValidationAttribute.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Extensions.Logging;
+using System;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Class)]
+    public sealed class EgressValidationAttribute : ActionFilterAttribute, IFilterFactory
+    {
+        public bool IsReusable => true;
+
+        IFilterMetadata IFilterFactory.CreateInstance(IServiceProvider serviceProvider)
+        {
+            var egressOutputOptions = serviceProvider.GetRequiredService<IEgressOutputOptions>();
+
+            EgressValidationActionFilter actionFilter = new EgressValidationActionFilter(egressOutputOptions);
+
+            return actionFilter;
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Class)]
+    public class EgressValidationUnhandledExceptionFilter : ActionFilterAttribute, IExceptionFilter
+    {
+        private readonly ILogger _logger;
+
+        public EgressValidationUnhandledExceptionFilter(ILogger<EgressValidationUnhandledExceptionFilter> logger)
+        {
+            _logger = logger;
+        }
+
+        public void OnException(ExceptionContext context)
+        {
+            if (context.Exception is EgressValidationActionFilter.EgressDisabledException egressException)
+            {
+                BadRequestObjectResult badRequestResult = new BadRequestObjectResult(egressException.Message);
```

Comment:
I'd write this as:
```
context.Result = new BadRequestObjectResult(
    egressException.ToProblemDetails(StatusCodes.Status400BadRequest));
```

There's no need to instantiate the BadRequestObjectResult using the exception message and then overwriting it on the next line by setting the Value property. If those are inlined together, then you don't need an explicit badRequestResult local variable either.

Alternatively, use [this](https://github.com/dotnet/dotnet-monitor/blob/bf6844e907c40b617eab6ef9c807065603678d51/src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagControllerExtensions.cs#L93) method (get ride of the `controller` parameter, it's not used anyway) since it does exactly what you want.